pkg/test/docker.go

```diff
@@ -43,23 +44,25 @@ func newDockerRunner() *dockerRunner {
 
 	pool.MaxWait = 2 * time.Minute
 
-	resources := make([]*dockertest.Resource, 0)
+	containers := make([]string, 0)
 
 	id := uuid.New().NewV4()
 
 	logger := mon.NewLogger().WithChannel("docker-runner")
 
 	return &dockerRunner{
-		pool:      pool,
-		resources: resources,
-		id:        id,
-		logger:    logger,
+		pool:       pool,
+		id:         id,
+		logger:     logger,
+		containers: containers,
 	}
 }
 
-func (d *dockerRunner) Run(name string, config containerConfig) (*dockertest.Resource, error) {
+func (d *dockerRunner) Run(name string, config containerConfig) error {
 	containerName := d.getContainerName(name)
 
+	d.markForCleanup(containerName)
```

Comment:
why would we do this before the container did even start successfully?

Reply:
to make sure it is cleaned up correctly. 
If the container failed to start, then it doesn't need to be removed, so we will ignore the error when trying to remove it.

Reply:
That sounds like a case to me that should never happen and also it does auto heal, because we do expire the container

Reply:
never mind, just had a look at dockertest. Due to the way they implemented `RunWithOptions` there is no better way.